src/workerd/api/sockets.h

```diff
@@ -11,7 +11,7 @@
 namespace workerd::api {
 
 struct SocketOptions {
-  jsg::Unimplemented tls; // TODO(later): TCP socket options need to be implemented.
+  bool tls;
```

Comment:
Feels a bit odd for this to be *just* a bool. We may end up expanding this out with more options in the future right? I wonder if it shouldn't be an empty struct for now? Hard to say for sure tho because that feels a bit odd also.

Reply:
I think `connect("example.org:443", { tls: true, tlsOptions: {...}})` would work well. But this something we should spec out properly. The API implemented here is non-final.

Reply:
Also worth keeping in mind, both Deno and Node.js have separate functions for connecting with or without TLS. I don't know if we want to emulate that but it's one of the options.

Node.js:
```
const net = require('net');
const tls = require('tls');
net.connect(...)
tls.connect(...)
```

Deno:
```
Deno.connect({ ... })
Deno.connectTls({ ... })
```